src/ORM/DataQuery.php

```diff
@@ -915,6 +917,19 @@ protected function joinHasManyRelation(
             return;
         }
 
+        if (class_exists(Versioned::class)) {
```

Comment:
@ScopeyNZ I decided to go for a different approach as extension points seems like overkill here. This code now checks if `Versioned` is available.